spec/helpers/appstore_project_helper.rb

```diff
@@ -0,0 +1,214 @@
+# this is a duplicate of the appstore project helper - we should delete it when we merge repositories
+
+require 'gooddata'
+require 'json'
+require_relative 'constants'
+
+module GoodData
+  module Helpers
+    # Creates a testing project and loads it with data
+    class AppstoreProjectHelper
+      attr_reader :project, :client
+
+      class << self
+        def create(opts = { client: GoodData.connection })
+          new_project = opts[:client].create_project(opts)
+          AppstoreProjectHelper.new(new_project, opts)
+        end
+
+        def suffix
+          hostname = Socket.gethostname
+          timestamp = DateTime.now.strftime('%Y%m%d%H%M%S')
+          suffix = "#{hostname}_#{timestamp}"
+          segment_name_forbidden_chars = /[^a-zA-Z0-9_\\-]+/
+          suffix.scan(segment_name_forbidden_chars).each do |forbidden_characters|
+            suffix.gsub!(forbidden_characters, '_')
+          end
+          suffix
+        end
+      end
+
+      def initialize(project, opts = { client: GoodData.connection })
+        @client = opts[:client]
+        @project = project if project.instance_of? GoodData::Project
+        @project = client.projects(project) if project.instance_of? String
+        GoodData.logger.info "Project ID: #{@project.obj_id}"
+        raise 'ProjectHelper::initialize the parameter "project" '\
+            'must be GoodData::Project or String.' unless @project
+      end
+
+      def create_ldm
+        json = File.read('./spec/data/integration_model.json')
+        blueprint = GoodData::Model::ProjectBlueprint.from_json(json)
+        project.update_from_blueprint(blueprint)
+      end
+
+      def load_data
+        GoodData.with_project @project.pid do |project|
+          GoodData::Model.upload_data(
+            './spec/data/integration_policies.csv',
+            project.blueprint,
+            DATASET_IDENTIFIER
+          )
+        end
+      end
+
+      def add_user_group
+        @project.add_user_group(:name => USER_GROUP_NAME, :description => 'My Test Description')
+      end
+
+      def associate_output_stage(ads, opts = {})
+        @project.create_output_stage(
+          ads,
+          project: @project,
+          client: client,
+          client_id: opts[:client_id],
+          output_stage_prefix: opts[:output_stage_prefix]
+        )
+      end
+
+      def deploy_processes(ads)
+        GoodData.with_project @project.pid do |project|
+          cc_process = project.deploy_process(
```

Comment:
Refers to 'project' more than self (maybe move it to another class?) (FeatureEnvy)